File: 1383-maximum-performance-of-a-team/1383-maximum-performance-of-a-team.cpp
class Solution {
public:
    int maxPerformance(int n, vector<int>& speed, vector<int>& efficiency, int k) {
        
        
    vector<pair<int,int>> engs(n); 
    for(int i=0; i< n; i++) {
        engs[i] = { efficiency[i], speed[i]};
    }
    sort( rbegin(engs) , rend(engs)) ; 
    long ssum = 0 , ans = 0 ; 
    priority_queue <int, vector<int> , greater<int>> heap ; 
    for(auto& [e,s]: engs) { 
        heap.emplace(s); 
        ssum +=s ; 
        if ( heap.size() >k) { 
            ssum -= heap.top() ; 
            heap.pop() ; 
        }
        ans = max(ans ,ssum*e); 
    }
    return ans %(int)(1e9+7); 


    }
};

File: 1383-maximum-performance-of-a-team/1383-maximum-performance-of-a-team.py
class Solution:
    def maxPerformance(self, n: int, speed: List[int], efficiency: List[int], k: int) -> int:
        

        engs = list(zip(efficiency, speed))
        engs.sort(reverse = True) 
        heap = [] 
        ans = ssum = 0 
        for e ,s in engs:
            ssum += s
            heappush(heap, s)
            if len(heap) > k:
                ssum -= heappop(heap)
            ans = max(ans, ssum*e)
        return ans % (10**9 + 7) 